pkg/apis/pubsub/v1alpha1/topic_validation_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/knative/pkg/webhook"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	topicSpec = TopicSpec{
+		Secret: &corev1.SecretKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: "secret-name",
+			},
+			Key: "secret-key",
+		},
+		Project:            "my-eventing-project",
+		Topic:              "pubsub-topic",
+		ServiceAccountName: "service-account-name",
+	}
+)
+
+func TestTopicValidation(t *testing.T) {
```

Comment:
Can you add a test using an invalid propagationPolicy value?

Reply:
https://github.com/GoogleCloudPlatform/cloud-run-events/issues/40